api-client/src/test/java/com/xing/api/CallSpecTest.java

```diff
@@ -635,6 +651,35 @@ public void onFailure(Throwable t) {
         assertThat(t.getMessage()).contains("malformed JSON");
     }
 
+    @SuppressWarnings("unchecked") // We don't care about type safety at this point.
+    @Test
+    public void specThrowsIfUnauthorizedAsync() throws Exception {
+        server.enqueue(new MockResponse().setResponseCode(401));
+
+        CallSpec spec = builder(HttpMethod.GET, "/", false)
+              .responseAs(Object.class)
+              .build();
+
+        final AtomicReference<Throwable> responseRef = new AtomicReference<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+        spec.enqueue(new Callback() {
+            @Override
+            public void onResponse(Response response) {
+                fail("unexpected #onResponse() call");
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+                responseRef.set(t);
+                latch.countDown();
+            }
+        });
+
+        latch.await(2, TimeUnit.SECONDS);
```

Comment:
some idea for the future:
isn't there a better solution for waiting for the response? some optional callback would be nice